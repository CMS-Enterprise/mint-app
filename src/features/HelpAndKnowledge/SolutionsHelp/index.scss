.operational-solution-help {
    &__category {
        .usa-card__container {
            border-radius: 4px !important;
            padding-top: 1rem !important;
            padding-bottom: 1rem !important;
            height: 7rem !important;
            display: flex !important;
            justify-content: space-between !important;
        }
    }
}
