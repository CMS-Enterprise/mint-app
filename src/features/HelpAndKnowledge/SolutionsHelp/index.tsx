/*
Operational Solution portion of the MINT Help and Knowledge Center
Contains components for search, categories, and solutions cards
*/

import React, { useEffect, useMemo, useState } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import { GridContainer } from '@trussworks/react-uswds';
import classNames from 'classnames';
import { MtoCommonSolutionSubject } from 'gql/generated/graphql';

import Divider from 'components/Divider';
import PageLoading from 'components/PageLoading';
import useHelpSolution from 'hooks/useHelpSolutions';
import useModalSolutionState from 'hooks/useModalSolutionState';

import CategoryFooter from './_components/CategoryFooter';
import SolutionHelpCardGroup from './_components/SolutionHelpCardGroup';
import SolutionsHeader from './_components/SolutionsHeader';
import SolutionDetailsModal from './SolutionDetails/Modal';
import {
  HelpSolutionBaseType,
  HelpSolutionType,
  routeToEnumMap
} from './solutionsMap';

type OperationalSolutionsHelpProps = {
  className?: string;
};

// Return all solutions relevant to the current cateory
export const findCategoryMapByRouteParam = (
  categoryKey: MtoCommonSolutionSubject,
  solutions: HelpSolutionType[]
): HelpSolutionBaseType[] => {
  return solutions.filter(solution => {
    return solution.categories.includes(categoryKey);
  });
};

export const findSolutionByRouteParam = (
  route: string | null | undefined,
  solutions: HelpSolutionType[],
  isEnum: boolean = false
): HelpSolutionType | undefined => {
  if (!route) return undefined;
  const routeParam = isEnum ? route : routeToEnumMap[route];
  return [...solutions].find(solution => solution.key === routeParam);
};

// Query function to return solutions for matching name and acronym
export const searchSolutions = (
  query: string,
  solutions: HelpSolutionType[]
): HelpSolutionType[] => {
  const queryRegex = new RegExp(query.toLowerCase());
  return solutions.filter(
    solution =>
      solution.name.toLowerCase().includes(query.toLowerCase()) ||
      solution?.acronym?.toLowerCase().includes(query.toLowerCase()) ||
      solution.categories.some(e => queryRegex.test(e.replace('-', ' ')))
  );
};

const SolutionsHelp = ({ className }: OperationalSolutionsHelpProps) => {
  const location = useLocation();
  const navigate = useNavigate();

  const params = useMemo(
    () => new URLSearchParams(location.search),
    [location.search]
  );

  const category = params.get('category') as MtoCommonSolutionSubject;
  const page = params.get('page');
  const solutionEnumParam = params.get('solution-key');
  const modal = params.get('solution') || solutionEnumParam;

  if (!page) {
    params.set('page', '1');
    navigate(
      {
        search: params.toString()
      },
      {
        replace: true
      }
    );
  }

  const { helpSolutions, loading } = useHelpSolution();

  // Get the solution map details from solution route param
  const { prevPathname, selectedSolution } = useModalSolutionState();

  const [query, setQuery] = useState<string>('');
  const [resultsNum, setResultsNum] = useState<number>(0);

  const [querySolutions, setQuerySolutions] =
    useState<HelpSolutionBaseType[]>(helpSolutions);

  const fromModal: boolean =
    prevPathname.includes('solution=') ||
    prevPathname.includes('solution-key=');

  // Resets the query on route or category change
  // Also preserves the query/scroll when the modal is open/closed
  useEffect(() => {
    if (!page && location.pathname && (!category || (!modal && !fromModal))) {
      setQuery('');
      window.scrollTo(0, 0);
    }
    if (!query && !modal && !fromModal) {
      window.scrollTo(0, 0);
    }
  }, [page, location.pathname, category, modal, query, fromModal]);

  //  If no query, return all solutions, otherwise, matching query solutions
  useEffect(() => {
    if (query.trim()) {
      const paramsChange = new URLSearchParams(location.search);
      paramsChange.set('page', '1');
      navigate({ search: paramsChange.toString() });
      setQuerySolutions(searchSolutions(query, helpSolutions));
    } else {
      setQuerySolutions(helpSolutions);
    }
  }, [query, selectedSolution, helpSolutions, location.search, navigate]);

  // If viewing by category, render those solutions, otherwise render querySolutions
  const solutions = !category
    ? querySolutions
    : findCategoryMapByRouteParam(category, helpSolutions);

  if (loading) {
    return <PageLoading />;
  }

  return (
    <div className={classNames(className)}>
      {selectedSolution && (
        <SolutionDetailsModal
          solution={selectedSolution}
          openedFrom={prevPathname}
          closeRoute="/help-and-knowledge/operational-solutions"
        />
      )}

      <SolutionsHeader
        category={category}
        resultsNum={resultsNum}
        resultsMax={solutions.length}
        setQuery={setQuery}
        query={query}
      />

      <SolutionHelpCardGroup
        solutions={solutions}
        setResultsNum={setResultsNum}
        category={category}
      />

      <GridContainer className="margin-top-4">
        <Divider className="margin-top-6" />

        <CategoryFooter currentCategory={category} />
      </GridContainer>
    </div>
  );
};

export default SolutionsHelp;
