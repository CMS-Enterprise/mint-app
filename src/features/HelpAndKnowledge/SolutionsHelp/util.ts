import { tObject } from 'utils/translation';

import { AboutConfigType } from './SolutionDetails/Solutions/Generic/About';
import { TimelineConfigType } from './SolutionDetails/Solutions/Generic/Timeline';
import { HelpSolutionBaseType, HelpSolutionsType } from './solutionsMap';

export const timelineTranslationUtil = (solutionKey: string) =>
  tObject<keyof TimelineConfigType, any>(
    `helpAndKnowledge:solutions.${solutionKey}.timeline`
  );

export const aboutTranslationUtil = (solutionKey: string) =>
  tObject<keyof AboutConfigType, any>(
    `helpAndKnowledge:solutions.${solutionKey}.about`
  );

export const convertSolutionObjectToArray = (
  solutionMap: HelpSolutionsType
): HelpSolutionBaseType[] =>
  Object.values(solutionMap).map(solution => solution);
