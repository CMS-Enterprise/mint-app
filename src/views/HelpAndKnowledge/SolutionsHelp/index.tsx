/*
Operational Solution portion of the MINT Help and Knowledge Center
Contains components for search, categories, and solutions cards
*/

import React, { useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';
import { GridContainer } from '@trussworks/react-uswds';
import classNames from 'classnames';

import Divider from 'components/shared/Divider';
import OperationalSolutionCategories from 'data/operationalSolutionCategories';
import usePrevLocation from 'hooks/usePrevious';

import CategoryFooter from './_components/CategoryFooter';
import SolutionHelpCardGroup from './_components/SolutionHelpCardGroup';
import SolutionsHeader from './_components/SolutionsHeader';
import SolutionDetailsModal from './SolutionDetails/Modal';
import {
  helpSolutions,
  HelpSolutionType,
  operationalSolutionCategoryMap
} from './solutionsMap';

type OperationalSolutionsHelpProps = {
  className?: string;
};

// Return all solutions relevant to the current cateory
export const findCategoryMapByRoute = (
  route: string,
  solutions: HelpSolutionType[]
): HelpSolutionType[] => {
  const categoryKey: OperationalSolutionCategories | undefined =
    operationalSolutionCategoryMap[route];

  if (!categoryKey) return [];

  return [...solutions].filter(solution => {
    return solution.categories.includes(
      categoryKey as OperationalSolutionCategories
    );
  });
};

export const findSolutionByRoute = (
  route: string | null,
  solutions: HelpSolutionType[]
): HelpSolutionType | undefined => {
  if (!route) return undefined;
  return [...solutions].find(solution => solution.route === route);
};

// Query function to return solutions for matching name and acronym
export const searchSolutions = (
  query: string,
  solutions: HelpSolutionType[]
): HelpSolutionType[] => {
  return solutions.filter(
    solution =>
      solution.name.toLowerCase().includes(query.toLowerCase()) ||
      solution?.acronym?.toLowerCase().includes(query.toLowerCase())
  );
};

const SolutionsHelp = ({ className }: OperationalSolutionsHelpProps) => {
  const location = useLocation();

  const params = new URLSearchParams(location.search);

  const category = params.get('category');
  const solution = params.get('solution');
  const page = params.get('page');

  const prevLocation = usePrevLocation(location);
  const prevPathname = prevLocation?.pathname + (prevLocation?.search || '');

  const [query, setQuery] = useState<string>('');
  const [resultsNum, setResultsNum] = useState<number>(0);

  const [querySolutions, setQuerySolutions] = useState<HelpSolutionType[]>(
    helpSolutions
  );

  // Resets the query on route or category change
  // Also preserves the query/scroll when the modal is open/closed
  useEffect(() => {
    if (!page && location.pathname) {
      setQuery('');
      window.scrollTo(0, 0);
    }
  }, [page, location.pathname, category]);

  //  If no query, return all solutions, otherwise, matching query solutions
  useEffect(() => {
    if (query.trim()) {
      setQuerySolutions(searchSolutions(query, helpSolutions));
    } else {
      setQuerySolutions(helpSolutions);
    }
  }, [query, solution]);

  // If viewing by category, render those solutions, otherwise render querySolutions
  const solutions = !category
    ? querySolutions
    : findCategoryMapByRoute(category, helpSolutions);

  // Solution to render in modal
  const selectedSolution = findSolutionByRoute(solution, helpSolutions);

  return (
    <div className={classNames(className)}>
      {selectedSolution && (
        <SolutionDetailsModal
          solution={selectedSolution}
          openedFrom={prevPathname}
        />
      )}

      <SolutionsHeader
        category={category}
        resultsNum={resultsNum}
        resultsMax={solutions.length}
        setQuery={setQuery}
        query={query}
      />

      <SolutionHelpCardGroup
        solutions={solutions}
        setResultsNum={setResultsNum}
        category={category}
      />

      <GridContainer className="margin-top-4">
        <Divider className="margin-top-6" />

        <CategoryFooter />
      </GridContainer>
    </div>
  );
};

export default SolutionsHelp;
