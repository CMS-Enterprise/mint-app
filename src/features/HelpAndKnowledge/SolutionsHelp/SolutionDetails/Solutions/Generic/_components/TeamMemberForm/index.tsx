import React, { useEffect, useState } from 'react';
import { Controller, FormProvider, useForm } from 'react-hook-form';
import { Trans, useTranslation } from 'react-i18next';
import { toast } from 'react-toastify';
import {
  Button,
  Fieldset,
  Form,
  FormGroup,
  Icon,
  Label,
  TextInput,
  Tooltip
} from '@trussworks/react-uswds';
import classNames from 'classnames';
import { SolutionContactType } from 'features/HelpAndKnowledge/SolutionsHelp/solutionsMap';
import {
  useCreateMtoCommonSolutionUserContactMutation,
  useUpdateMtoCommonSolutionContactMutation
} from 'gql/generated/graphql';
import GetMTOSolutionContacts from 'gql/operations/ModelToOperations/GetMTOSolutionContacts';

import Alert from 'components/Alert';
import CheckboxField from 'components/CheckboxField';
import OktaUserSelect from 'components/OktaUserSelect';
import { useErrorMessage } from 'contexts/ErrorContext';
import useModalSolutionState from 'hooks/useModalSolutionState';
import dirtyInput from 'utils/formUtil';

import { TeamMemberModeType } from '../MailboxAndTeamMemberModal';

type UnwrapNullable<
  T extends Record<P, unknown> | null | undefined,
  P extends PropertyKey
> = T extends Record<P, unknown> ? T[P] : '';

export type TeamMemberFormValues = Pick<
  SolutionContactType,
  'name' | 'role' | 'isPrimary' | 'receiveEmails'
> & {
  userName: UnwrapNullable<SolutionContactType['userAccount'], 'username'>;
};

const TeamMemberForm = ({
  mode,
  closeModal,
  teamMember = {
    __typename: 'MTOCommonSolutionContact',
    id: 'not a real id',
    name: '',
    email: '',
    isTeam: false,
    isPrimary: false,
    role: '',
    receiveEmails: false,
    userAccount: {
      __typename: 'UserAccount',
      id: 'not a real userAccount id',
      username: ''
    }
  },
  setDisableButton
}: {
  mode: TeamMemberModeType;
  closeModal: () => void;
  teamMember?: SolutionContactType;
  setDisableButton: React.Dispatch<React.SetStateAction<boolean>>;
}) => {
  const { t: contactT } = useTranslation('mtoCommonSolutionContact');
  const { t: miscT } = useTranslation('mtoCommonSolutionContactMisc');

  const { selectedSolution } = useModalSolutionState();

  const methods = useForm<TeamMemberFormValues>({
    defaultValues: {
      userName: teamMember.userAccount?.username || '',
      name: teamMember.name,
      role: teamMember.role || '',
      isPrimary: teamMember.isPrimary,
      receiveEmails: teamMember.receiveEmails
    },
    mode: 'onChange'
  });

  const {
    control,
    handleSubmit,
    formState: { isSubmitting, isDirty },
    watch,
    setValue
  } = methods;

  const [create] = useCreateMtoCommonSolutionUserContactMutation({
    refetchQueries: [
      {
        query: GetMTOSolutionContacts
      }
    ]
  });

  const [update] = useUpdateMtoCommonSolutionContactMutation({
    refetchQueries: [
      {
        query: GetMTOSolutionContacts
      }
    ]
  });

  const [mutationError, setMutationError] = useState<
    'duplicate' | 'generic' | null
  >(null);

  const isAddMode = mode === 'addTeamMember';
  const isEditMode = mode === 'editTeamMember';

  const disabledSubmitBtn =
    !watch('userName') || !watch('role') || isSubmitting || !isDirty;

  const { setErrorMeta } = useErrorMessage();

  useEffect(() => {
    setDisableButton(disabledSubmitBtn);
  }, [setDisableButton, disabledSubmitBtn]);

  const onSubmit = (formData: TeamMemberFormValues) => {
    if (!selectedSolution) {
      setMutationError('generic');
      return;
    }

    const { role, isPrimary, receiveEmails } = dirtyInput(teamMember, formData);

    setErrorMeta({
      overrideMessage: `${formData.userName} is already added to this solution and cannot be added again. Please edit the existing entry.`
    });

    const promise = isAddMode
      ? create({
          variables: {
            key: selectedSolution.key,
            userName: formData.userName,
            role: formData.role || '',
            isPrimary: formData.isPrimary,
            receiveEmails: formData.isPrimary ? true : formData.receiveEmails
          }
        })
      : update({
          variables: {
            id: teamMember.id,
            input: {
              role,
              isPrimary,
              receiveEmails: isPrimary ? true : receiveEmails
            }
          }
        });
    promise.then(response => {
      if (!response?.errors) {
        toast.success(
          <Alert type="success" isClosable={false}>
            <Trans
              i18nKey={`mtoCommonSolutionContactMisc:${mode}.success`}
              values={{
                contact: formData.name || teamMember.name
              }}
              components={{
                bold: <span className="text-bold" />
              }}
            />
          </Alert>,
          {
            autoClose: 3000,
            hideProgressBar: false
          }
        );
        closeModal();
      }
    });
  };

  return (
    <FormProvider {...methods}>
      <Form
        className="maxw-none padding-bottom-10"
        data-testid="team-member-form"
        id="team-member-form"
        onSubmit={handleSubmit(onSubmit)}
      >
        {mutationError !== null && (
          <Alert
            type="error"
            slim
            headingLevel="h1"
            className="margin-bottom-2"
          >
            {mutationError === 'generic' ? (
              miscT(`${mode}.error`)
            ) : (
              <Trans
                i18nKey="mtoCommonSolutionContactMisc:duplicateError"
                values={{
                  contact: methods.getValues('name')
                }}
                components={{
                  bold: <span className="text-bold" />
                }}
              />
            )}
          </Alert>
        )}
        <Fieldset disabled={!selectedSolution} style={{ minWidth: '100%' }}>
          <Controller
            name="userName"
            control={control}
            render={({ field: { ref, ...field } }) => (
              <FormGroup className="margin-top-0 margin-bottom-2">
                <Label
                  htmlFor="team-member-name"
                  className="mint-body-normal maxw-none margin-bottom-1"
                  requiredMarker
                >
                  {contactT('name.label')}
                </Label>
                <span className="text-base-dark">
                  {contactT('name.sublabel')}
                </span>

                <OktaUserSelect
                  id="team-member-name"
                  name="team-member-name"
                  ariaLabelledBy="label-team-member-name"
                  ariaDescribedBy="hint-team-member-name"
                  value={{
                    username: teamMember.userAccount?.username || '',
                    displayName: teamMember.name,
                    email: teamMember.email
                  }}
                  onChange={oktaUser => {
                    setValue('name', oktaUser ? oktaUser.displayName : '');
                    setValue('userName', oktaUser ? oktaUser.username : '');
                  }}
                  className={classNames({
                    'disabled-input': isEditMode
                  })}
                  disabled={isEditMode}
                />
              </FormGroup>
            )}
          />

          <Controller
            name="role"
            control={control}
            rules={{
              required: true,
              validate: value => value !== ''
            }}
            render={({ field: { ref, ...field } }) => (
              <FormGroup className="margin-top-0">
                <Label
                  htmlFor="team-member-role"
                  className="mint-body-normal maxw-none margin-bottom-1"
                  requiredMarker
                >
                  {contactT('role.label')}
                </Label>

                <TextInput
                  type="text"
                  {...field}
                  id="team-member-role"
                  data-testid="team-member-role"
                  value={field.value || ''}
                />
              </FormGroup>
            )}
          />

          <Controller
            name="isPrimary"
            control={control}
            render={({ field: { ref, ...field } }) => (
              <FormGroup className="margin-top-0 margin-bottom-2">
                <CheckboxField
                  {...field}
                  id="isPrimary"
                  testid="isPrimary"
                  label={contactT('isPrimary.label')}
                  subLabel={
                    isAddMode
                      ? contactT('isPrimary.sublabel')
                      : miscT('editTeamMember.primaryPocSubLabel')
                  }
                  checked={Boolean(field.value)}
                  value="true"
                  onBlur={field.onBlur}
                  onChange={e => {
                    field.onChange(e.target.checked);
                  }}
                  disabled={teamMember.isPrimary}
                  icon={
                    <Tooltip
                      label={contactT('isPrimary.questionTooltip')}
                      position="top"
                      className="bg-white padding-0 text-base-dark"
                      style={{ gap: '0.25rem' }}
                    >
                      <Icon.Info className="minw-4 text-base-light margin-left-neg-05 margin-bottom-neg-2px" />
                    </Tooltip>
                  }
                />
              </FormGroup>
            )}
          />

          <Controller
            name="receiveEmails"
            control={control}
            render={({ field: { ref, ...field } }) => (
              <FormGroup className="margin-top-0 margin-bottom-2">
                <CheckboxField
                  {...field}
                  id="receiveEmails"
                  testid="receiveEmails"
                  label={contactT('receiveEmails.label')}
                  subLabel={contactT('receiveEmails.sublabel')}
                  checked={Boolean(field.value) || watch('isPrimary')}
                  value="true"
                  onBlur={field.onBlur}
                  onChange={e => {
                    field.onChange(e.target.checked);
                  }}
                  disabled={watch('isPrimary')}
                />
              </FormGroup>
            )}
          />
        </Fieldset>
        <Alert
          type="info"
          headingLevel="h1"
          slim
          className="margin-top-0 margin-bottom-2"
          hidden={!watch('isPrimary') && !watch('receiveEmails')}
        >
          <Trans
            i18nKey="mtoCommonSolutionContactMisc:alert"
            components={{
              milestoneLibrary: (
                <Button
                  type="button"
                  className="margin-top-0"
                  unstyled
                  onClick={() => {}}
                >
                  {' '}
                </Button>
              )
            }}
          />
        </Alert>
      </Form>
    </FormProvider>
  );
};

export default TeamMemberForm;
