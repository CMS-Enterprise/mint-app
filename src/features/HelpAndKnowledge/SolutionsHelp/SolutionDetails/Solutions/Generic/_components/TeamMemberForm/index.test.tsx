import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, waitFor } from '@testing-library/react';
import { SolutionContactType } from 'features/HelpAndKnowledge/SolutionsHelp/solutionsMap';
import { possibleSolutionsMock } from 'tests/mock/mto';

import MessageProvider from 'contexts/MessageContext';

import TeamMemberForm from './index';

const mocks = [...possibleSolutionsMock];

const team: SolutionContactType = {
  __typename: 'MTOCommonSolutionContact',
  id: '123',
  name: 'Blue Bonnet',
  email: 'memberemail@email.com',
  mailboxTitle: '',
  mailboxAddress: '',
  isTeam: false,
  isPrimary: false,
  role: 'Project Lead',
  receiveEmails: false
};

describe('Team member point of contact form', () => {
  it('should render team member info accordingly', async () => {
    const { getByTestId, getByText } = render(
      <MemoryRouter
        initialEntries={[
          '/help-and-knowledge/operational-solutions/solutions?solution=accountable-care-organization&section=points-of-contact'
        ]}
      >
        <MockedProvider mocks={mocks} addTypename={false}>
          <MessageProvider>
            <Route path="/help-and-knowledge/operational-solutions">
              <TeamMemberForm
                mode="editTeamMember"
                closeModal={() => {}}
                teamMember={team}
                setDisableButton={() => {}}
              />
            </Route>
          </MessageProvider>
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(
        getByText('This field searches CMSâ€™ EUA database.')
      ).toBeInTheDocument();
      expect(getByTestId('team-member-role')).toHaveValue('Project Lead');
      expect(getByTestId('isPrimary')).not.toBeChecked();
      expect(getByTestId('receiveEmails')).not.toBeChecked();
    });
  });

  it('should render team mailbox info accordingly', async () => {
    const { getByTestId } = render(
      <MemoryRouter
        initialEntries={[
          '/help-and-knowledge/operational-solutions/solutions?solution=accountable-care-organization&section=points-of-contact'
        ]}
      >
        <MockedProvider mocks={mocks} addTypename={false}>
          <MessageProvider>
            <Route path="/help-and-knowledge/operational-solutions">
              <TeamMemberForm
                mode="editTeamMember"
                closeModal={() => {}}
                teamMember={{ ...team, isPrimary: true, receiveEmails: true }}
                setDisableButton={() => {}}
              />
            </Route>
          </MessageProvider>
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(getByTestId('isPrimary')).toBeChecked();
      expect(getByTestId('receiveEmails')).toBeChecked();
    });
  });

  it('should matches snapshot', async () => {
    const { asFragment } = render(
      <MemoryRouter
        initialEntries={[
          '/help-and-knowledge/operational-solutions/solutions?solution=accountable-care-organization&section=points-of-contact'
        ]}
      >
        <MockedProvider mocks={mocks} addTypename={false}>
          <MessageProvider>
            <Route path="/help-and-knowledge/operational-solutions">
              <TeamMemberForm
                mode="addTeamMember"
                closeModal={() => {}}
                teamMember={team}
                setDisableButton={() => {}}
              />
            </Route>
          </MessageProvider>
        </MockedProvider>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
