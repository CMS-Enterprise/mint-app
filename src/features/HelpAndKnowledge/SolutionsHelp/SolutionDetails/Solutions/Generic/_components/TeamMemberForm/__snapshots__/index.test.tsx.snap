// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Team member point of contact form > should matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="usa-form maxw-none padding-bottom-10"
    data-testid="team-member-form"
    id="team-member-form"
  >
    <fieldset
      class="usa-fieldset"
      data-testid="fieldset"
      style="min-width: 100%;"
    >
      <div
        class="usa-form-group margin-top-0 margin-bottom-2"
        data-testid="formGroup"
      >
        <label
          class="usa-label mint-body-normal maxw-none margin-bottom-1"
          data-testid="label"
          for="team-member-name"
        >
          Name 
          <abbr
            class="usa-hint usa-hint--required"
            title="required"
          >
            *
          </abbr>
        </label>
        <span
          class="text-base-dark"
        >
          This field searches CMS’ EUA database.
        </span>
        <div
          class="margin-top-1 cedar-contact-select maxw-none usa-combo-box cedar-contact-select__loading css-b62m3t-container"
          id="team-member-name"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-team-member-name-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
            role="log"
          />
          <div
            class="cedar-contact-select__control cedar-contact-select__control--menu-is-open css-79rxjr-control"
          >
            <div
              class="cedar-contact-select__value-container cedar-contact-select__value-container--has-value css-1fdsijx-ValueContainer"
            >
              <div
                class="cedar-contact-select__input-container css-166bipr-Input"
                data-value="Blue Bonnet (memberemail@email.com)"
              >
                <input
                  aria-activedescendant=""
                  aria-autocomplete="list"
                  aria-controls="react-select-team-member-name-listbox"
                  aria-expanded="true"
                  aria-haspopup="true"
                  aria-labelledby="label-team-member-name"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="cedar-contact-select__input"
                  id="react-select-team-member-name-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value="Blue Bonnet (memberemail@email.com)"
                />
              </div>
            </div>
            <div
              class="cedar-contact-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                aria-valuetext="Loading"
                class="mint-spinner mint-spinner--small margin-right-1"
                data-testid="spinner"
                role="progressbar"
              />
              <button
                aria-label="Clear selection"
                class="usa-button--unstyled"
                id="clear-selection"
                tabindex="0"
                type="button"
              >
                <div
                  aria-hidden="true"
                  class="cedar-contact-select__indicator cedar-contact-select__clear-indicator css-1h8bgg-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                    />
                  </svg>
                </div>
              </button>
              <span
                class="cedar-contact-select__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="cedar-contact-select__indicator cedar-contact-select__dropdown-indicator css-1h8bgg-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="cedar-contact-select__menu css-1v1oxpf-menu"
          >
            <div
              aria-multiselectable="false"
              class="cedar-contact-select__menu-list css-bgzggo-MenuList"
              id="react-select-team-member-name-listbox"
              role="listbox"
            >
              <div
                class="cedar-contact-select__menu-notice cedar-contact-select__menu-notice--no-options css-14zharz-NoOptionsMessage"
              >
                <button
                  aria-label="No results found"
                  class="usa-button--unstyled text-no-underline text-black"
                  id="no-results"
                  tabindex="0"
                  type="button"
                >
                  No results found
                </button>
              </div>
            </div>
          </div>
          <input
            name="team-member-name"
            type="hidden"
            value="[object Object]"
          />
        </div>
      </div>
      <div
        class="usa-form-group margin-top-0"
        data-testid="formGroup"
      >
        <label
          class="usa-label mint-body-normal maxw-none margin-bottom-1"
          data-testid="label"
          for="team-member-role"
        >
          Role 
          <abbr
            class="usa-hint usa-hint--required"
            title="required"
          >
            *
          </abbr>
        </label>
        <input
          class="usa-input"
          data-testid="team-member-role"
          id="team-member-role"
          name="role"
          type="text"
          value="Project Lead"
        />
      </div>
      <div
        class="usa-form-group margin-top-0 margin-bottom-2"
        data-testid="formGroup"
      >
        <div
          class="mint-checkbox usa-checkbox"
        >
          <input
            class="usa-checkbox__input"
            data-testid="isPrimary"
            id="isPrimary"
            name="isPrimary"
            type="checkbox"
            value="true"
          />
          <label
            class="usa-checkbox__label"
            for="isPrimary"
          >
            Primary point of contact
          </label>
          <span
            class="usa-tooltip"
            data-testid="tooltipWrapper"
          >
            <button
              aria-describedby="tooltip-«r2»"
              class="usa-button usa-tooltip__trigger bg-white padding-0 text-base-dark"
              data-testid="triggerElement"
              style="gap: 0.25rem;"
              tabindex="0"
              title=""
              type="button"
            >
              <svg
                class="usa-icon minw-4 text-base-light margin-left-neg-05 margin-bottom-neg-2px"
                focusable="false"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                />
              </svg>
            </button>
            <span
              aria-hidden="true"
              class="usa-tooltip__body"
              data-testid="tooltipBody"
              id="tooltip-«r2»"
              role="tooltip"
              title="The primary point of contact is the main person or mailbox that should be contacted with questions about this IT system or solution. They will receive update emails from MINT when a model chooses to use their system or solution."
            >
              The primary point of contact is the main person or mailbox that should be contacted with questions about this IT system or solution. They will receive update emails from MINT when a model chooses to use their system or solution.
            </span>
          </span>
          <p
            class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs text-pre-line"
          >
            If you check this box, it will replace any existing primary point of contact, but will not remove the existing contact from the overall list.
          </p>
        </div>
      </div>
      <div
        class="usa-form-group margin-top-0 margin-bottom-2"
        data-testid="formGroup"
      >
        <div
          class="mint-checkbox usa-checkbox"
        >
          <input
            class="usa-checkbox__input"
            data-testid="receiveEmails"
            id="receiveEmails"
            name="receiveEmails"
            type="checkbox"
            value="true"
          />
          <label
            class="usa-checkbox__label"
            for="receiveEmails"
          >
            Receive emails?
          </label>
          <p
            class="margin-y-1 margin-left-4 line-height-body-3 font-body-2xs text-pre-line"
          >
            If you check this box, this user or mailbox will receive update notifications from MINT when models select this operational solution. Primary points of contact will always receive notifications.
          </p>
        </div>
      </div>
    </fieldset>
    <div
      class="usa-alert usa-alert--info usa-alert--slim flex margin-top-0 margin-bottom-2"
      data-testid="alert"
      hidden=""
    >
      <div
        class="usa-alert__body"
      >
        <p
          class="usa-alert__text"
        >
          <span>
            This point of contact will receive notifications when this solution is selected. Please make sure this individual should receive these notifications before you proceed.
          </span>
        </p>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
